File: src/Domain/Entities/ScheduleTask.cs
namespace Domain.Entities;

public sealed class ScheduleTask
{
    public int Id { get; init; }
    public string Name { get; init; } = string.Empty;
    public int Duration { get; init; }

    public int EarlyStart { get; set; }
    public int EarlyFinish { get; set; }
    public int LateStart { get; set; }
    public int LateFinish { get; set; }
    public int TotalFloat { get; set; }
    public int FreeFloat { get; set; }
    public bool IsCritical { get; set; }
    public int ProjectId { get; set; }

    public decimal Budget { get; init; }
    public double PercentComplete { get; set; }
    public decimal ActualCost { get; set; }
}

File: web/src/app/core/models/schedule-task.ts
export interface ScheduleTask {
  id: number;
  name: string;
  duration: number;
  projectId: number;
  earlyStart: number;
  earlyFinish: number;
  lateStart: number;
  lateFinish: number;
  totalFloat: number;
  freeFloat: number;
  isCritical: boolean;
  budget: number;
  percentComplete: number;
  actualCost: number;
}
